MetaMask/metamask-extension
# Fix: GasTiming: should only accept string types

## Explanation

It appears that `ui/components/app/gas-details-item/gas-details-item.js` is passing both number and string types to the GasTiming component. We should only be accepting the string type here. 

Audit of all instances of the GasTiming component:
- `ui/components/app/edit-gas-display/edit-gas-display.component.js`
- `ui/components/app/gas-details-item/gas-details-item.js`
- `ui/pages/confirm-transaction-base/confirm-transaction-base.component.js` 

## Description

Fixes https://github.com/MetaMask/metamask-extension/issues/14764

## Screenshots/Screencaps

<img width="1554" alt="Screenshot 2022-05-07 at 9 36 26 PM" src="https://user-images.githubusercontent.com/20778143/167510597-29c226ab-77a4-4917-b61e-7d1c5f8e861a.png">

## Manual Testing Steps

1. Issue a transaction
2. Confirm GasTiming is working as expected 
3. Observe the following errors shown in the screenshot above are not present with these changes
